# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/vm/Interpreter.cpp
# Commit: 7a3a4207c55b
# Full Hash: 7a3a4207c55b27d6a73783c0021cfed2368bdbb8
# Author: Jan de Mooij <jdemooij@mozilla.com>
# Date: 2019-12-06 21:48:33
# Regressor Bug: 1601897
# File Overlap Count: 3
# Description:
#   Bug 1601897 part 2 - Move patchBreaks and addTryNote calls to LoopControl::emitLoopEnd. r=arai
#   
#   This ensures all loops handle this the same way.
#   
#   For JSTRY_FOR_IN some places asserted the try note ends at the
# ==============================================================================

diff -r 7a9ab8423e85 -r 7a3a4207c55b js/src/vm/Interpreter.cpp
--- a/js/src/vm/Interpreter.cpp	Fri Dec 06 11:08:02 2019 +0000
+++ b/js/src/vm/Interpreter.cpp	Fri Dec 06 11:08:04 2019 +0000
@@ -1192,9 +1192,7 @@
 
       case JSTRY_FOR_IN: {
         /* This is similar to JSOP_ENDITER in the interpreter loop. */
-        DebugOnly<jsbytecode*> pc =
-            regs.fp()->script()->offsetToPC(tn->start + tn->length);
-        MOZ_ASSERT(JSOp(*pc) == JSOP_ENDITER);
+        MOZ_ASSERT(tn->stackDepth <= regs.stackDepth());
         Value* sp = regs.spForStackDepth(tn->stackDepth);
         JSObject* obj = &sp[-1].toObject();
         CloseIterator(obj);
